### ISSUE TITLE
Timestamps not preserved

### ISSUE THREAD

**Issue by PF4Public** [actions/checkout#364]
Is it by design, that timestamps are not preserved upon checkout or do I need to make it implicit somehow?

**Comment by marusak** [comment]
Seems that adding
```
        with:
          fetch-depth: 0
```
solves this problem.

**Comment by PF4Public** [comment]
> Seems that adding
> 
> ```
>         with:
>           fetch-depth: 0
> ```
> 
> solves this problem.

I tried, but this didn't help unfortunately.

**Comment by marusak** [comment]
are you using `v2`?
```
uses: actions/checkout@v2
```

**Comment by PF4Public** [comment]
I can't remember, because since then, I used another approach.
But if you can confirm, that with `fetch-depth: 0` timestamps are preserved, I will close this issue.

**Comment by mitchell-psh** [comment]
So, I tried changing fetch-depth to 0, while using v2, but this didn't preserve timestamps. On delving into it, I realized that, if you guys were attempting something similar to me, then all 3 of us are dumb. Git purposely does not retain timestamps on checkout. If you guys were attempting something else, then I'm leaving this here for future souls traveling down my forlorn path.

See https://git.wiki.kernel.org/index.php/Git_FAQ#Why_isn.27t_Git_preserving_modification_time_on_files.3F
